GoogleForCreators/web-stories-wp
# Make sure the saved colour presets are always unique

## Context
Some oddities were happening around the saved colors, the reason for these bugs is having multiple colors with the same value. It's not clear exactly how this happened in the QA environment, however, I guess it might have something to do with multiple people testing the global colors at the same time.


## Summary
This PR ensures that when the color presets are saved, these are always unique. It also ensures that when the colors are loaded, no matter what's in the DB, the colors displayed in the editor are always unique.


## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions
Do this on `main` first without checking out this PR first:
1. Download [this file](https://drive.google.com/file/d/1z44fZv9EQrg3AbnKnUIf-gAs8M2r-UY1/view?usp=sharing)
2. Create a new story and click `Shift + Option + Command + J` at the same time
3. The debug window opens. Choose "base64", paste the content of the downloaded file into the area to replace what's there and click "Load"
4. Save the story.
5. Verify that now you have multiple grey colours in the "All stories" and when you open the colour picker and switch between the Current and the All colours, more grey colours start appearing.
6. Check out this PR.
7. Verify that now there are no multiple identical grey colours under "All Stories"
8. Verify that now switching between the two modes in the colour picker does not make the pseudo colours appear.


### QA






This PR can be tested by following these steps:

1. Please follow the steps above under the "Testing Instructions" -- ping Miina if there are issues with following the steps.

### UAT







This PR can be tested by following these steps:

1. Ensure there are some Saved colours ander the "All stories" and some under the "Current story" section
2. Now open the colour picker for a shape or for the page background.
3. Switch between the "All stories" and "Current story".
4. Verify no pseudo colours appear.

## Checklist




---



Fixes #6700 
Fixes #6701
